segmentation fault when scanning a CycloneDX bom.json without dependencies

When having a `bom.json` file generated from a maven project without dependencies, osv-scanner fails.

Tested with osv-scanner version 1.2 and 1.3

### How to reproduce

1. save the following `pom.xml` in an empty folder
```
<project xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>text</groupId>
    <artifactId>test</artifactId>
    <packaging>jar</packaging>
    <version>1.0</version>
    <dependencies>
    </dependencies>
</project>
```

2. Generate CycloneDX SBOM file 
```
mvn -ntp -B org.cyclonedx:cyclonedx-maven-plugin:makeAggregateBom
```

3. run osv-scanner 
```
osv-scanner --sbom target/bom.json
```

4. you can see the following error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7d7817]

goroutine 1 [running]:
github.com/google/osv-scanner/internal/sbom.(*CycloneDX).enumeratePackages(0xc000128758?, 0x30?, 0xc0001b6c00)
	/home/runner/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/internal/sbom/cyclonedx.go:25 +0x37
github.com/google/osv-scanner/internal/sbom.(*CycloneDX).GetPackages(0xc00017e5e8?, {0xb8c5c8?, 0xc000128758}, 0x5?)
	/home/runner/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/internal/sbom/cyclonedx.go:50 +0x217
github.com/google/osv-scanner/pkg/osvscanner.scanSBOMFile(0xc0001ce700, 0xc000122b40, {0xc00017e5a0, 0x50})
	/home/runner/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/pkg/osvscanner/osvscanner.go:235 +0x246
github.com/google/osv-scanner/pkg/osvscanner.DoScan({{0x0, 0x0, 0x0}, {0xc0001af7f0, 0x1, 0x1}, {0xf76918, 0x0, 0x0}, {0x0, ...}, ...}, ...)
	/home/runner/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/pkg/osvscanner/osvscanner.go:424 +0x4a5
main.run.func3(0xc0001ce180)
	/home/runner/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/cmd/osv-scanner/main.go:108 +0x458
github.com/urfave/cli/v2.(*Command).Run(0xc000192420, 0xc0001ce180, {0xc00011a150, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.24.4/command.go:273 +0x9eb
github.com/urfave/cli/v2.(*App).RunContext(0xc000210000, {0xb8dcb8?, 0xc00012e028}, {0xc00011a150, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.24.4/app.go:332 +0x616
github.com/urfave/cli/v2.(*App).Run(...)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.24.4/app.go:309
main.run({0xc00011a150, 0x3, 0x3}, {0xb89d00?, 0xc000128008}, {0xb89d00?, 0xc000128010})
	/home/runner/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/cmd/osv-scanner/main.go:127 +0x9b8
main.main()
	/home/runner/go/pkg/mod/github.com/google/osv-scanner@v1.2.0/cmd/osv-scanner/main.go:153 +0x46
```

--- a/internal/sbom/cyclonedx.go
+++ b/internal/sbom/cyclonedx.go
@@ -69,6 +69,10 @@ func (c *CycloneDX) enumerateComponents(components []cyclonedx.Component, callba
 }
 
 func (c *CycloneDX) enumeratePackages(bom *cyclonedx.BOM, callback func(Identifier) error) error {
+	if bom.Components == nil {
+		return nil
+	}
+
 	return c.enumerateComponents(*bom.Components, callback)
 }
 

--- a/internal/sbom/cyclonedx_test.go
+++ b/internal/sbom/cyclonedx_test.go
@@ -1,0 +1,58 @@
+package sbom_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/osv-scanner/internal/sbom"
+)
+
+func runCycloneGetPackages(t *testing.T, bomFile string, want []sbom.Identifier) {
+	t.Helper()
+
+	f, err := os.Open("fixtures/" + bomFile)
+	if err != nil {
+		t.Fatalf("Failed to read fixture file: %v", err)
+	}
+
+	got := []sbom.Identifier{}
+	callback := func(id sbom.Identifier) error {
+		got = append(got, id)
+		return nil
+	}
+
+	cdx := &sbom.CycloneDX{}
+	err = cdx.GetPackages(f, callback)
+	if err != nil {
+		t.Errorf("GetPackages returned an error: %v", err)
+	}
+
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("GetPackages() returned an unexpected result (-want, +got):\n%s", diff)
+	}
+}
+
+func TestCycloneDXGetPackages(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		bomFile     string
+		identifiers []sbom.Identifier
+	}{
+		{
+			bomFile: "cyclonedx.json",
+			identifiers: []sbom.Identifier{
+				{PURL: "pkg:maven/org.hdrhistogram/HdrHistogram@2.1.12"},
+				{PURL: "pkg:maven/org.apache.logging.log4j/log4j-core@2.16.0"},
+			},
+		},
+		{
+			bomFile:     "cyclonedx-empty.json",
+			identifiers: []sbom.Identifier{},
+		},
+	}
+
+	for _, tt := range tests {
+		runCycloneGetPackages(t, tt.bomFile, tt.identifiers)
+	}
+}

--- a/internal/sbom/cyclonedx_test.go
+++ b/internal/sbom/cyclonedx_test.go
@@ -2,6 +2,7 @@ package sbom_test
 
 import (
 	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
@@ -11,10 +12,11 @@ import (
 func runCycloneGetPackages(t *testing.T, bomFile string, want []sbom.Identifier) {
 	t.Helper()
 
-	f, err := os.Open("fixtures/" + bomFile)
+	f, err := os.Open(filepath.Join("fixtures", bomFile))
 	if err != nil {
 		t.Fatalf("Failed to read fixture file: %v", err)
 	}
+	defer f.Close()
 
 	got := []sbom.Identifier{}
 	callback := func(id sbom.Identifier) error {

--- a/internal/sbom/fixtures/cyclonedx-empty.json
+++ b/internal/sbom/fixtures/cyclonedx-empty.json
@@ -1,0 +1,5 @@
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.4",
+  "version": 1
+}

--- a/internal/sbom/fixtures/cyclonedx.json
+++ b/internal/sbom/fixtures/cyclonedx.json
@@ -1,0 +1,23 @@
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.4",
+  "version": 1,
+  "components": [
+    {
+      "type": "container",
+      "name": "/target.tar",
+      "components": [
+        {
+          "type": "library",
+          "name": "HdrHistogram",
+          "purl": "pkg:maven/org.hdrhistogram/HdrHistogram@2.1.12"
+        }
+      ]
+    },
+    {
+      "type": "library",
+      "name": "Apache Log4j Core",
+      "purl": "pkg:maven/org.apache.logging.log4j/log4j-core@2.16.0"
+    }
+  ]
+}

